### PR TITLE
Basic python3 support.

### LINKED ISSUE #28: Python 3 Upgrade
Now that synapse supports python 3, what about sygnal?

Looking at the dependencies, flask supports python 3 on the currently pinned version. However  gevent (now pinned at 1.0.1) will need to be at least 1.1. pushbaby also needs an update, but this shouldn't be difficult as it only depends on gevent.

### PR DESCRIPTION
**Pull request by evilham**
This has been tested to work fine with FB / GCM, apns may need further
changes.

Fixes #28.

### CODE REVIEW

File: sygnal/__init__.py [hunk 1]
```diff
@@ -15,7 +15,10 @@
 # limitations under the License.
 
 
-import ConfigParser
+try:
+    import ConfigParser as configparser
+except ImportError:
+    import configparser
```
Comment (erikjohnston) [inline sygnal/__init__.py]: You can use `six` to do this: `from six.moves import configparser`

File: sygnal/__init__.py [hunk 2]
```diff
@@ -171,13 +174,16 @@ class ClientError(Exception):
 
 
 def parse_config():
-    cfg = ConfigParser.SafeConfigParser(CONFIG_DEFAULTS)
+    if hasattr(configparser, 'SafeConfigParser'):
+        cfg = configparser.SafeConfigParser(CONFIG_DEFAULTS)
+    else:
+        cfg = configparser.ConfigParser(CONFIG_DEFAULTS)
```
Comment (erikjohnston) [inline sygnal/__init__.py]: The `six` configparse always has `SafeConfigParser

File: sygnal/db.py [hunk 3]
```diff
@@ -17,15 +17,18 @@
 import sqlite3
 import logging
 import threading
-import Queue
+try:
+    import Queue as queue
+except ImportError:
+    import queue
```
Comment (erikjohnston) [inline sygnal/db.py]: `from six.moves import queue`

**Comment by evilham** [comment]
Wonderful, I wasn't sure if pulling six was a desired thing, and since there wasn't much that needed touching...

Will take a go at it soon.

**Comment by erikjohnston** [comment]
Ah, cool, yeah I don't think pulling in `six` is a big deal tbh.

**Comment by evilham** [comment]
This would be ready for another review.

**Comment by erikjohnston** [comment]
Awesome! Thank you

**Pull request merged**